### PR TITLE
Adds use_repr to model_addition_debugger_context

### LINKED ISSUE #37983: Support saving tensors to a file in Model addition debuggers
### Feature request

Current state: [`model_addition_debugger_context` context manager](https://huggingface.co/docs/transformers/v4.51.3/en/internal/model_debugging_utils#model-addition-debuggers) write out a `FULL_TENSORS.json` file containing `repr()`-ized version of the relevant tensors. 

Request: Add a `use_repr: bool` option to `model_addition_debugger_context` that works exactly as above when `True` but saves the full tensors to disk using SafeTensors when set to `False`.

### Motivation

Some model adders, including me in my role as the Hugging Face lead for Gemma, release models on many platforms and need to perform numerical validations of logits across those implementations to ensure consistency and quality. The `repr()`-ized version in the current version of `model_addition_debugger_context` is insufficient to support these sorts of validations. Dumping the tensors to the JSON file will induce _a lot_ of bloat, so the best and most flexible option is to save the tensors to discrete files and reference those files from the `FULL_TENSORS.json` file. 

### Your contribution

I am working on an implementation of this feature.

### PR DESCRIPTION
**Pull request by RyanMullins**
# What does this PR do?

Adds a `use_repr: bool` option to `model_addition_debugger_context` to support saving full tensors to disk instead of `repr()`-ized versions in the JSON file.

Fixes #37983

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@molbap @ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
LGTM! 🤗

**Review by molbap — APPROVED** [review]
LGTM @RyanMullins , thank you! I left a minor comment, and something valuable would be to update `model_debugging_utils.md` with the new option (with the large output size disclaimer)

**Comment by RyanMullins** [comment]
Was preparing for and at Google IO today. Will update the PR and send it your way tomorrow.
